diff --git a/vendor/magento/module-catalog/Setup/CategorySetup.php b/vendor/magento/module-catalog/Setup/CategorySetup.php
index e8a7a7a..d894bd0 100644
--- a/vendor/magento/module-catalog/Setup/CategorySetup.php
+++ b/vendor/magento/module-catalog/Setup/CategorySetup.php
@@ -25,6 +25,16 @@ class CategorySetup extends EavSetup
     private $categoryFactory;
 
     /**
+     * This should be set explicitly
+     */
+    const CATEGORY_ENTITY_TYPE_ID = 3;
+
+    /**
+     * This should be set explicitly
+     */
+    const CATALOG_PRODUCT_ENTITY_TYPE_ID = 4;
+
+    /**
      * Init
      *
      * @param ModuleDataSetupInterface $setup
@@ -66,6 +76,7 @@ class CategorySetup extends EavSetup
     {
         return [
             'catalog_category' => [
+                'entity_type_id' => self::CATEGORY_ENTITY_TYPE_ID,
                 'entity_model' => 'Magento\Catalog\Model\ResourceModel\Category',
                 'attribute_model' => 'Magento\Catalog\Model\ResourceModel\Eav\Attribute',
                 'table' => 'catalog_category_entity',
@@ -334,6 +345,7 @@ class CategorySetup extends EavSetup
                 ],
             ],
             'catalog_product' => [
+                'entity_type_id' => self::CATALOG_PRODUCT_ENTITY_TYPE_ID,
                 'entity_model' => 'Magento\Catalog\Model\ResourceModel\Product',
                 'attribute_model' => 'Magento\Catalog\Model\ResourceModel\Eav\Attribute',
                 'table' => 'catalog_product_entity',
diff --git a/vendor/magento/module-customer/Setup/CustomerSetup.php b/vendor/magento/module-customer/Setup/CustomerSetup.php
index dc99278..57d0034 100644
--- a/vendor/magento/module-customer/Setup/CustomerSetup.php
+++ b/vendor/magento/module-customer/Setup/CustomerSetup.php
@@ -126,6 +126,7 @@ class CustomerSetup extends EavSetup
     {
         $entities = [
             'customer' => [
+                'entity_type_id' => \Magento\Customer\Api\CustomerMetadataInterface::ATTRIBUTE_SET_ID_CUSTOMER,
                 'entity_model' => 'Magento\Customer\Model\ResourceModel\Customer',
                 'attribute_model' => 'Magento\Customer\Model\Attribute',
                 'table' => 'customer_entity',
@@ -338,6 +339,7 @@ class CustomerSetup extends EavSetup
                 ],
             ],
             'customer_address' => [
+                'entity_type_id' => \Magento\Customer\Api\AddressMetadataInterface::ATTRIBUTE_SET_ID_ADDRESS,
                 'entity_model' => 'Magento\Customer\Model\ResourceModel\Address',
                 'attribute_model' => 'Magento\Customer\Model\Attribute',
                 'table' => 'customer_address_entity',
diff --git a/vendor/magento/module-eav/Setup/EavSetup.php b/vendor/magento/module-eav/Setup/EavSetup.php
index a53ed60..9ed8ffc 100644
--- a/vendor/magento/module-eav/Setup/EavSetup.php
+++ b/vendor/magento/module-eav/Setup/EavSetup.php
@@ -192,6 +192,9 @@ class EavSetup
             'additional_attribute_table' => $this->_getValue($params, 'additional_attribute_table'),
             'entity_attribute_collection' => $this->_getValue($params, 'entity_attribute_collection'),
         ];
+        if (isset($params['entity_type_id'])) {
+            $data['entity_type_id'] = $params['entity_type_id'];
+        }
 
         if ($this->getEntityType($code, 'entity_type_id')) {
             $this->updateEntityType($code, $data);
@@ -199,7 +202,11 @@ class EavSetup
             $this->setup->getConnection()->insert($this->setup->getTable('eav_entity_type'), $data);
         }
 
-        $this->addAttributeSet($code, $this->_defaultAttributeSetName);
+        if (isset($params['entity_type_id'])) {
+            $this->addAttributeSet($code, $this->_defaultAttributeSetName, null, $params['entity_type_id']);
+        } else {
+            $this->addAttributeSet($code, $this->_defaultAttributeSetName);
+        }
         $this->addAttributeGroup($code, $this->_defaultGroupName, $this->_generalGroupName);
 
         return $this;
@@ -310,9 +317,10 @@ class EavSetup
      * @param int|string $entityTypeId
      * @param string $name
      * @param int $sortOrder
+     * @param int $setId
      * @return $this
      */
-    public function addAttributeSet($entityTypeId, $name, $sortOrder = null)
+    public function addAttributeSet($entityTypeId, $name, $sortOrder = null, $setId = null)
     {
         $data = [
             'entity_type_id' => $this->getEntityTypeId($entityTypeId),
@@ -320,6 +328,10 @@ class EavSetup
             'sort_order' => $this->getAttributeSetSortOrder($entityTypeId, $sortOrder),
         ];
 
+        if ($setId !== null) {
+            $data['attribute_set_id'] = $setId;
+        }
+
         $setId = $this->getAttributeSet($entityTypeId, $name, 'attribute_set_id');
         if ($setId) {
             $this->updateAttributeSet($entityTypeId, $setId, $data);
diff --git a/vendor/magento/module-sales/Setup/SalesSetup.php b/vendor/magento/module-sales/Setup/SalesSetup.php
index d69147b..3c72ce0 100644
--- a/vendor/magento/module-sales/Setup/SalesSetup.php
+++ b/vendor/magento/module-sales/Setup/SalesSetup.php
@@ -19,6 +19,26 @@ use Magento\Framework\Setup\ModuleDataSetupInterface;
 class SalesSetup extends \Magento\Eav\Setup\EavSetup
 {
     /**
+     * This should be set explicitly
+     */
+    const ORDER_ENTITY_TYPE_ID = 5;
+
+    /**
+     * This should be set explicitly
+     */
+    const INVOICE_PRODUCT_ENTITY_TYPE_ID = 6;
+
+    /**
+     * This should be set explicitly
+     */
+    const CREDITMEMO_PRODUCT_ENTITY_TYPE_ID = 7;
+
+    /**
+     * This should be set explicitly
+     */
+    const SHIPMENT_PRODUCT_ENTITY_TYPE_ID = 8;
+
+    /**
      * @var ScopeConfigInterface
      */
     protected $config;
@@ -214,6 +234,7 @@ class SalesSetup extends \Magento\Eav\Setup\EavSetup
     {
         $entities = [
             'order' => [
+                'entity_type_id' => self::ORDER_ENTITY_TYPE_ID,
                 'entity_model' => 'Magento\Sales\Model\ResourceModel\Order',
                 'table' => 'sales_order',
                 'increment_model' => 'Magento\Eav\Model\Entity\Increment\NumericValue',
@@ -221,6 +242,7 @@ class SalesSetup extends \Magento\Eav\Setup\EavSetup
                 'attributes' => [],
             ],
             'invoice' => [
+                'entity_type_id' => self::INVOICE_PRODUCT_ENTITY_TYPE_ID,
                 'entity_model' => 'Magento\Sales\Model\ResourceModel\Order\Invoice',
                 'table' => 'sales_invoice',
                 'increment_model' => 'Magento\Eav\Model\Entity\Increment\NumericValue',
@@ -228,6 +250,7 @@ class SalesSetup extends \Magento\Eav\Setup\EavSetup
                 'attributes' => [],
             ],
             'creditmemo' => [
+                'entity_type_id' => self::CREDITMEMO_PRODUCT_ENTITY_TYPE_ID,
                 'entity_model' => 'Magento\Sales\Model\ResourceModel\Order\Creditmemo',
                 'table' => 'sales_creditmemo',
                 'increment_model' => 'Magento\Eav\Model\Entity\Increment\NumericValue',
@@ -235,6 +258,7 @@ class SalesSetup extends \Magento\Eav\Setup\EavSetup
                 'attributes' => [],
             ],
             'shipment' => [
+                'entity_type_id' => self::SHIPMENT_PRODUCT_ENTITY_TYPE_ID,
                 'entity_model' => 'Magento\Sales\Model\ResourceModel\Order\Shipment',
                 'table' => 'sales_shipment',
                 'increment_model' => 'Magento\Eav\Model\Entity\Increment\NumericValue',
