MAGETWO-63020
diff --Naur a/vendor/magento/framework/View/Design/Theme/FlyweightFactory.php b/vendor/magento/framework/View/Design/Theme/FlyweightFactory.php
--- a/vendor/magento/framework/View/Design/Theme/FlyweightFactory.php  2017-01-06 20:41:07.000000000 +0000
+++ b/vendor/magento/framework/View/Design/Theme/FlyweightFactory.php  2017-01-06 20:42:33.000000000 +0000
@@ -61,7 +61,7 @@
         } else {
             $themeModel = $this->_loadByPath($themeKey, $area);
         }
-        if (!$themeModel->getId()) {
+        if (!$themeModel->getCode()) {
             throw new \LogicException("Unable to load theme by specified key: '{$themeKey}'");
         }
         $this->_addTheme($themeModel);
