MAGETWO-63032
diff -Naur a/vendor/magento/framework/Console/Cli.php b/vendor/magento/framework/Console/Cli.php
--- a/vendor/magento/framework/Console/Cli.php
+++ b/vendor/magento/framework/Console/Cli.php
@@ -165,9 +165,7 @@
             $omProvider = $this->serviceManager->get(ObjectManagerProvider::class);
             $omProvider->setObjectManager($this->objectManager);
         } catch (FileSystemException $exception) {
-            $this->writeGenerationDirectoryReadError();

-            exit(static::RETURN_FAILURE);
         }
     }

@@ -195,9 +193,7 @@
             $state->getMode() !== State::MODE_PRODUCTION
             && !$generationDirectoryAccess->check()
         ) {
-            $this->writeGenerationDirectoryReadError();

-            exit(static::RETURN_FAILURE);
         }
     }

@@ -223,29 +219,12 @@
             try {
                 $compilerPreparation->handleCompilerEnvironment();
             } catch (FileSystemException $e) {
-                $this->writeGenerationDirectoryReadError();

-                exit(static::RETURN_FAILURE);
             }
         }
     }

-    /**
-     * Writes read error to console.
-     *
-     * @return void
-     */
-    private function writeGenerationDirectoryReadError()
-    {
-        $output = new \Symfony\Component\Console\Output\ConsoleOutput();
-        $output->writeln(
-            '<error>'
-            . 'Command line user does not have read and write permissions on '
-            . $this->getDefaultDirectoryPath(DirectoryList::GENERATED_CODE) . ' directory. '
-            . 'Please address this issue before using Magento command line.'
-            . '</error>'
-        );
-    }
+

     /**
      * Retrieves vendor commands.
