diff --git a/vendor/magento/module-cron/Console/Command/CronCommand.php b/vendor/magento/module-cron/Console/Command/CronCommand.php
index 78bbb2329f8..d238cb9ac34 100644
--- a/vendor/magento/module-cron/Console/Command/CronCommand.php
+++ b/vendor/magento/module-cron/Console/Command/CronCommand.php
@@ -14,6 +14,7 @@ use Magento\Framework\App\ObjectManagerFactory;
 use Magento\Store\Model\Store;
 use Magento\Store\Model\StoreManager;
 use Magento\Cron\Observer\ProcessCronQueueObserver;
+use Magento\Framework\App\DeploymentConfig;
 use Magento\Framework\Console\Cli;
 use Magento\Framework\Shell\ComplexParameter;
 
@@ -35,13 +36,20 @@ class CronCommand extends Command
     private $objectManagerFactory;
 
     /**
-     * Constructor
+     * Application deployment configuration
      *
+     * @var DeploymentConfig
+     */
+    private $deploymentConfig;
+
+    /**
      * @param ObjectManagerFactory $objectManagerFactory
+     * @param DeploymentConfig $deploymentConfig
      */
-    public function __construct(ObjectManagerFactory $objectManagerFactory)
+    public function __construct(ObjectManagerFactory $objectManagerFactory, DeploymentConfig $deploymentConfig)
     {
         $this->objectManagerFactory = $objectManagerFactory;
+        $this->deploymentConfig = $deploymentConfig;
         parent::__construct();
     }
 
@@ -71,10 +79,16 @@ class CronCommand extends Command
     }
 
     /**
+     * Runs cron jobs if cron is not disabled in Magento configurations
+     *
      * {@inheritdoc}
      */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
+        if (!$this->deploymentConfig->get('cron/enabled', 1)) {
+            $output->writeln('<info>' . 'Cron is disabled. Jobs were not run.' . '</info>');
+            return;
+        }
         $omParams = $_SERVER;
         $omParams[StoreManager::PARAM_RUN_CODE] = 'admin';
         $omParams[Store::CUSTOM_ENTRY_POINT_PARAM] = true;
