diff -Nuar a/setup/src/Magento/Setup/Console/Command/UpgradeCommand.php b/setup/src/Magento/Setup/Console/Command/UpgradeCommand.php
--- a/setup/src/Magento/Setup/Console/Command/UpgradeCommand.php
+++ b/setup/src/Magento/Setup/Console/Command/UpgradeCommand.php
@@ -77,6 +77,7 @@ class UpgradeCommand extends AbstractSetupCommand
     protected function execute(InputInterface $input, OutputInterface $output)
     {
         try {
+            $resultCode = \Magento\Framework\Console\Cli::RETURN_SUCCESS;
             $keepGenerated = $input->getOption(self::INPUT_KEY_KEEP_GENERATED);
             $installer = $this->installerFactory->create(new ConsoleLogger($output));
             $installer->updateModulesSequence($keepGenerated);
@@ -87,10 +88,10 @@ class UpgradeCommand extends AbstractSetupCommand
                 $importConfigCommand = $this->getApplication()->find(ConfigImportCommand::COMMAND_NAME);
                 $arrayInput = new ArrayInput([]);
                 $arrayInput->setInteractive($input->isInteractive());
-                $importConfigCommand->run($arrayInput, $output);
+                $resultCode = $importConfigCommand->run($arrayInput, $output);
             }

-            if (!$keepGenerated) {
+            if ($resultCode !== \Magento\Framework\Console\Cli::RETURN_FAILURE && !$keepGenerated) {
                 $output->writeln(
                     '<info>Please re-run Magento compile command. Use the command "setup:di:compile"</info>'
                 );
@@ -100,6 +101,6 @@ class UpgradeCommand extends AbstractSetupCommand
             return \Magento\Framework\Console\Cli::RETURN_FAILURE;
         }

-        return \Magento\Framework\Console\Cli::RETURN_SUCCESS;
+        return $resultCode;
     }
 }
