diff -Nuar a/vendor/magento/module-message-queue/Model/Cron/ConsumersRunner/PidConsumerManager.php b/vendor/magento/module-message-queue/Model/Cron/ConsumersRunner/PidConsumerManager.php
--- a/vendor/magento/module-message-queue/Model/Cron/ConsumersRunner/PidConsumerManager.php
+++ b/vendor/magento/module-message-queue/Model/Cron/ConsumersRunner/PidConsumerManager.php
@@ -119,7 +119,7 @@ class PidConsumerManager
      */
     public function getPid($consumerName)
     {
-        $pidFile = $consumerName . static::PID_FILE_EXT;
+        $pidFile = $this->getPidFileName($consumerName);
         /** @var WriteInterface $directory */
         $directory = $this->filesystem->getDirectoryWrite(DirectoryList::VAR_DIR);
 
@@ -138,7 +138,7 @@ class PidConsumerManager
      */
     public function getPidFilePath($consumerName)
     {
-        return $this->directoryList->getPath(DirectoryList::VAR_DIR) . '/' . $consumerName . static::PID_FILE_EXT;
+        return $this->directoryList->getPath(DirectoryList::VAR_DIR) . '/' . $this->getPidFileName($consumerName);
     }
 
     /**
@@ -152,4 +152,17 @@ class PidConsumerManager
         $file->write(function_exists('posix_getpid') ? posix_getpid() : getmypid());
         $file->close();
     }
+
+    /**
+     * Returns default file name with PID by consumers name
+     *
+     * @param string $consumerName The consumers name
+     * @return string The file name with PID
+     */
+    private function getPidFileName($consumerName)
+    {
+        $sanitizedHostname = preg_replace('/[^a-z0-9]/i', '', gethostname());
+
+        return $consumerName . '-' . $sanitizedHostname . static::PID_FILE_EXT;
+    }
 }
