diff -Nuar a/vendor/colinmollenhour/credis/Client.php b/vendor/colinmollenhour/credis/Client.php
index afbc85d..8368b32 100755
--- a/vendor/colinmollenhour/credis/Client.php
+++ b/vendor/colinmollenhour/credis/Client.php
@@ -1017,6 +1017,7 @@ class Credis_Client {
                     } else {
                         $this->isMulti = TRUE;
                         $this->redisMulti = call_user_func_array(array($this->redis, $name), $args);
+                        return $this;
                     }
                 }
                 else if($name == 'exec' || $name == 'discard') {
