diff -Naur a/vendor/magento/module-catalog/Model/ResourceModel/Product/Collection.php b/vendor/magento/module-catalog/Model/ResourceModel/Product/Collection.php
--- a/vendor/magento/module-catalog/Model/ResourceModel/Product/Collection.php
+++ b/vendor/magento/module-catalog/Model/ResourceModel/Product/Collection.php
@@ -1584,6 +1584,8 @@ class Collection extends \Magento\Catalog\Model\ResourceModel\Collection\Abstrac
         } else {
             return parent::addAttributeToFilter($attribute, $condition, $joinType);
         }
+
+        return $this;
     }

     /**
