diff -Nuar a/vendor/magento/framework/Mail/Message.php b/vendor/magento/framework/Mail/Message.php
index 36a0e6a..fad0910 100644
--- a/vendor/magento/framework/Mail/Message.php
+++ b/vendor/magento/framework/Mail/Message.php
@@ -15,6 +15,7 @@ class Message extends \Zend_Mail implements MessageInterface
     public function __construct($charset = 'utf-8')
     {
         parent::__construct($charset);
+        $this->setHeaderEncoding(\Zend_Mime::ENCODING_BASE64);
     }
 
     /**
