diff --git a/vendor/magento/module-catalog-import-export/Model/Import/Uploader.php b/vendor/magento/module-catalog-import-export/Model/Import/Uploader.php
index 4dfde38..fcc544f 100644
--- a/vendor/magento/module-catalog-import-export/Model/Import/Uploader.php
+++ b/vendor/magento/module-catalog-import-export/Model/Import/Uploader.php
@@ -155,7 +155,8 @@ class Uploader extends \Magento\MediaStorage\Model\File\Uploader
 
         $filePath = $this->_directory->getRelativePath($this->getTmpDir() . '/' . $fileName);
         $this->_setUploadFile($filePath);
-        $result = $this->save($this->getDestDir());
+        $destDir = $this->_directory->getAbsolutePath($this->getDestDir());
+        $result = $this->save($destDir);
         $result['name'] = self::getCorrectFileName($result['name']);
         return $result;
     }
@@ -305,11 +306,10 @@ class Uploader extends \Magento\MediaStorage\Model\File\Uploader
             $tmpRealPath = $this->_directory->getDriver()->getRealPath(
                 $this->_directory->getAbsolutePath($tmpPath)
             );
-            $destinationRealPath = $this->_directory->getDriver()->getRealPath(
-                $this->_directory->getAbsolutePath($destPath)
-            );
+            $destinationRealPath = $this->_directory->getDriver()->getRealPath($destPath);
+            $relativeDestPath = $this->_directory->getRelativePath($destPath);
             $isSameFile = $tmpRealPath === $destinationRealPath;
-            return $isSameFile ?: $this->_directory->copyFile($tmpPath, $destPath);
+            return $isSameFile ?: $this->_directory->copyFile($tmpPath, $relativeDestPath);
         } else {
             return false;
         }
