diff --git a/app/etc/NonComposerComponentRegistration.php b/app/NonComposerComponentRegistration.php
rename from app/etc/NonComposerComponentRegistration.php
rename to app/NonComposerComponentRegistration.php
index bd3289d..400670c
--- a/app/etc/NonComposerComponentRegistration.php
+++ b/app/NonComposerComponentRegistration.php
@@ -4,12 +4,12 @@
  * See COPYING.txt for license details.
  */
 
-$pathList[] = dirname(__DIR__) . '/code/*/*/cli_commands.php';
-$pathList[] = dirname(__DIR__) . '/code/*/*/registration.php';
-$pathList[] = dirname(__DIR__) . '/design/*/*/*/registration.php';
-$pathList[] = dirname(__DIR__) . '/i18n/*/*/registration.php';
-$pathList[] = dirname(dirname(__DIR__)) . '/lib/internal/*/*/registration.php';
-$pathList[] = dirname(dirname(__DIR__)) . '/lib/internal/*/*/*/registration.php';
+$pathList[] = __DIR__ . '/code/*/*/cli_commands.php';
+$pathList[] = __DIR__ . '/code/*/*/registration.php';
+$pathList[] = __DIR__ . '/design/*/*/*/registration.php';
+$pathList[] = __DIR__ . '/i18n/*/*/registration.php';
+$pathList[] = dirname(__DIR__) . '/lib/internal/*/*/registration.php';
+$pathList[] = dirname(__DIR__) . '/lib/internal/*/*/*/registration.php';
 foreach ($pathList as $path) {
     // Sorting is disabled intentionally for performance improvement
     $files = glob($path, GLOB_NOSORT);
diff --git a/composer.json b/composer.json
index 417f2aa..c8d3cbb 100644
--- a/composer.json
+++ b/composer.json
@@ -227,7 +227,7 @@
             "": "app/code/"
         },
         "files": [
-            "app/etc/NonComposerComponentRegistration.php"
+            "app/NonComposerComponentRegistration.php"
         ]
     },
     "autoload-dev": {
