MAGECLOUD-589 MAGETWO-64955
diff -Naur a/vendor/magento/framework/View/Asset/File/FallbackContext.php b/vendor/magento/framework/View/Asset/File/FallbackContext.php
--- a/vendor/magento/framework/View/Asset/File/FallbackContext.php	2016-10-15 12:38:21.595266000 +0000
+++ b/vendor/magento/framework/View/Asset/File/FallbackContext.php	2016-10-15 12:39:13.587266000 +0000
@@ -103,6 +103,6 @@
      */
     public function getConfigPath()
     {
-        return $this->getPath() . ($this->isSecure ? '/' . self::SECURE_PATH : '');
+        return $this->getPath();
     }
 }

