--- a/vendor/magento/module-translation/Model/Json/PreProcessor.php
+++ b/vendor/magento/module-translation/Model/Json/PreProcessor.php
@@ -77,7 +77,7 @@ class PreProcessor implements PreProcessorInterface
             if ($context instanceof FallbackContext) {
                 $themePath = $context->getThemePath();
                 $areaCode = $context->getAreaCode();
-                $this->translate->setLocale($context->getLocale());
+                $this->translate->setLocale($context->getLocale())->loadData($areaCode);
             }

             $area = $this->areaList->getArea($areaCode);
