diff --git a/vendor/magento/framework/View/Design/FileResolution/Fallback/TemplateFile.php b/vendor/magento/framework/View/Design/FileResolution/Fallback/TemplateFile.php
index 09f87d878..5ef71afcc 100644
--- a/vendor/magento/framework/View/Design/FileResolution/Fallback/TemplateFile.php
+++ b/vendor/magento/framework/View/Design/FileResolution/Fallback/TemplateFile.php
@@ -107,9 +107,12 @@ class TemplateFile extends File
      */
     private function getMinifiedTemplateInProduction($template)
     {
-        if ($this->deploymentConfig->getConfigData(
-            ConfigOptionsListConstants::CONFIG_PATH_SCD_ON_DEMAND_IN_PRODUCTION
-        )) {
+        $forceMinified = $this->deploymentConfig->getConfigData(
+                ConfigOptionsListConstants::CONFIG_PATH_SCD_ON_DEMAND_IN_PRODUCTION
+            )
+            || $this->deploymentConfig->getConfigData('force_html_minification');
+
+        if ($forceMinified) {
             return $this->templateMinifier->getMinified($template);
         }
         return $this->templateMinifier->getPathToMinified($template);
