diff -Nuar a/vendor/magento/module-require-js/Model/FileManager.php b/vendor/magento/module-require-js/Model/FileManager.php
--- a/vendor/magento/module-require-js/Model/FileManager.php
+++ b/vendor/magento/module-require-js/Model/FileManager.php
@@ -164,6 +164,9 @@
             }
 
             foreach ($libDir->read($bundleDir) as $bundleFile) {
+                if (pathinfo($bundleFile, PATHINFO_EXTENSION) !== 'js') {
+                    continue;
+                }
                 $relPath = $libDir->getRelativePath($bundleFile);
                 $bundles[] = $this->assetRepo->createArbitrary($relPath, '');
             }
