MAGECLOUD-589 MAGETWO-64955
diff -Naur a/vendor/magento/framework/View/Asset/File/FallbackContext.php b/vendor/magento/framework/View/Asset/File/FallbackContext.php
--- a/vendor/magento/framework/View/Asset/File/FallbackContext.php	2016-10-15 12:38:21.595266000 +0000
+++ b/vendor/magento/framework/View/Asset/File/FallbackContext.php	2016-10-15 12:39:13.587266000 +0000
@@ -103,6 +103,6 @@
      */
     public function getConfigPath()
     {
-        return $this->getPath() . ($this->isSecure ? '/' . self::SECURE_PATH : '');
+        return $this->getPath();
     }
 }

diff -Nuar a/vendor/magento/framework/View/Asset/Repository.php b/vendor/magento/framework/View/Asset/Repository.php
--- a/vendor/magento/framework/View/Asset/Repository.php	2017-02-16 16:50:18.000000000 +0000
+++ b/vendor/magento/framework/View/Asset/Repository.php	2017-02-22 14:29:40.000000000 +0000
@@ -269,8 +269,7 @@
                     'baseUrl' => $url,
                     'areaType' => $area,
                     'themePath' => $themePath,
-                    'localeCode' => $locale,
-                    'isSecure' => $isSecure
+                    'localeCode' => $locale
                 ]
             );
         }
