Patch regarding MAGETWO-56675.
diff -Nuar a/vendor/magento/framework/Interception/PluginList/PluginList.php b/vendor/magento/framework/Interception/PluginList/PluginList.php
--- a/vendor/magento/framework/Interception/PluginList/PluginList.php
+++ b/vendor/magento/framework/Interception/PluginList/PluginList.php
@@ -277,6 +277,7 @@ class PluginList extends Scoped implements InterceptionPluginList
                 $virtualTypes = [];
                 foreach ($this->_scopePriorityScheme as $scopeCode) {
                     if (false == isset($this->_loadedScopes[$scopeCode])) {
+                        $this->_loadedScopes[$scopeCode] = true;
                         $data = $this->_reader->read($scopeCode);
                         unset($data['preferences']);
                         if (!count($data)) {
@@ -285,7 +286,6 @@ class PluginList extends Scoped implements InterceptionPluginList
                         $this->_inherited = [];
                         $this->_processed = [];
                         $this->merge($data);
-                        $this->_loadedScopes[$scopeCode] = true;
                         foreach ($data as $class => $config) {
                             if (isset($config['type'])) {
                                 $virtualTypes[] = $class;
